EditImprovements/Edit.patch

```diff
@@ -0,0 +1,285 @@
+       END;
+-      IF Text.hasCar THEN pos := Text.carloc.pos ELSE pos := 0 END;
+-      len := Text.text.len;
+-      Texts.OpenReader(R, Text.text, pos);
+-      Forward(M, R, buf); pos := pos + M;
++      IF F.hasCar THEN pos := F.carloc.pos ELSE pos := 0 END;
++      len := F.text.len;
++      Texts.OpenReader(R, F.text, pos);
++      Forward(M, R, buf); INC(pos, M);
+       j := M;
+       REPEAT DEC(j) UNTIL (j < 0) OR (buf[j] # pat[j]);
+       WHILE (j >= 0) & (pos < len) DO
+@@ -164,69 +240,14 @@
+         REPEAT DEC(j) UNTIL (j < 0) OR (buf[j] # pat[j])
+       END ;
+       IF j < 0 THEN
+-        TextFrames.RemoveSelection(Text); TextFrames.RemoveCaret(Text);
+-        Oberon.RemoveMarks(Text.X, Text.Y, Text.W, Text.H);
+-        TextFrames.Show(Text, pos - 300); Oberon.PassFocus(V);
+-        TextFrames.SetCaret(Text, pos)
++        TextFrames.RemoveSelection(F); TextFrames.RemoveCaret(F);
++        Oberon.RemoveMarks(F.X, F.Y, F.W, F.H);
++        ShowPos(F, pos);
++        Oberon.PassFocus(V);
++        TextFrames.SetCaret(F, pos)
+       END
+     END
+   END Search;
+ 
+-  PROCEDURE Locate*;
+-    VAR Text: TextFrames.Frame;
+-      T: Texts.Text; S: Texts.Scanner;
+-      V: Viewers.Viewer;
+-      beg, end, time: LONGINT;
+-  BEGIN
+-    V := Oberon.FocusViewer;
+-    IF (V.dsc # NIL) & (V.dsc.next IS TextFrames.Frame) THEN
+-      Text := V.dsc.next(TextFrames.Frame);
+-      Oberon.GetSelection(T, beg, end, time);
+-      IF time >= 0 THEN
+-        Texts.OpenScanner(S, T, beg);
+-        REPEAT Texts.Scan(S) UNTIL (S.class >= Texts.Int); (*skip names*)
+-        IF S.class = Texts.Int THEN
+-          TextFrames.RemoveSelection(Text);
+-          TextFrames.RemoveCaret(Text);
+-          Oberon.RemoveMarks(Text.X, Text.Y, Text.W, Text.H);
+-          TextFrames.Show(Text, Max(0, S.i - 200));
+-          Oberon.PassFocus(V);
+-          TextFrames.SetCaret(Text, S.i)
+-        END
+-      END
+-    END
+-  END Locate;
+-
+-  PROCEDURE Recall*;
+-    VAR Menu, Main: Display.Frame;
+-      buf: Texts.Buffer;
+-      V: Viewers.Viewer;
+-      pos: LONGINT;
+-      M: TextFrames.Frame;
+-  BEGIN V := Oberon.FocusViewer;
+-    IF (V # NIL) & (V IS MenuViewers.Viewer) THEN
+-      Menu := V.dsc; Main := V.dsc.next;
+-      IF Main IS TextFrames.Frame THEN
+-        M := Main(TextFrames.Frame);
+-        IF M.hasCar THEN
+-          TextFrames.Recall(buf);
+-          pos := M.carloc.pos + buf.len;
+-          Texts.Insert(M.text, M.carloc.pos, buf);
+-          TextFrames.SetCaret(M, pos)
+-        END
+-      ELSIF Menu IS TextFrames.Frame THEN
+-        M := Menu(TextFrames.Frame);
+-        IF M.hasCar THEN
+-          TextFrames.Recall(buf);
+-          pos := M.carloc.pos + buf.len;
+-          Texts.Insert(M.text, M.carloc.pos, buf);
+-          TextFrames.SetCaret(M, pos)
+-        END
+-      END
+-    END
+-  END Recall;
+-
+ BEGIN Texts.OpenWriter(W)
+-END Edit.
+-
+-
++END Edit.
```

Comment:
Whitespace change only...

Reply:
Yup, at the end. Is this bad?

Reply:
Aesthetics. Beauty is in the eye of the beholder.